static/js/components/EmploymentForm.js

```diff
@@ -0,0 +1,224 @@
+  componentWillMount () {
+    const { profile, setWorkHistoryEdit } = this.props;
+    if ( _.isArray(profile.work_history) && _.isEmpty(profile.work_history) ) {
+      setWorkHistoryEdit(false);
+    }
+  }
+
+  saveWorkHistoryEntry = () => {
+    saveAndContinue.call(this).then(() => {
+      this.closeWorkDialog();
+    });
+  }
+
+  toggleWorkHistoryEdit = () => {
+    const { ui, setWorkHistoryEdit } = this.props;
+    setWorkHistoryEdit(!ui.workHistoryEdit);
+  }
+
+  closeWorkDialog = () => {
+    const { setWorkDialogVisibility, clearProfileEdit } = this.props;
+    setWorkDialogVisibility(false);
+    clearProfileEdit();
+  }
+
+  addWorkHistoryEntry = () => {
+    const {
+      updateProfile,
+      profile,
+      setWorkDialogIndex,
+      setWorkDialogVisibility,
+    } = this.props;
+    let clone = Object.assign({}, profile, {
+      work_history: profile.work_history.concat(generateNewWorkHistory())
+    });
+    updateProfile(clone);
+    setWorkDialogIndex(clone.work_history.length - 1);
+    setWorkDialogVisibility(true);
+  }
+
+  deleteWorkHistoryEntry = index => {
+    const { saveProfile, profile } = this.props;
+    let clone = _.cloneDeep(profile);
+    clone['work_history'].splice(index, 1);
+    saveProfile(clone);
+  }
+
+  editWorkHistoryForm () {
+    const { ui } = this.props;
+    let keySet = (key) => ['work_history', ui.workDialogIndex, key];
+    return (
+      <Grid className="profile-tab-grid">
+        <Cell col={12} className="profile-form-title">
+          Add Employment
+        </Cell>
+        <Cell col={12}>
+          {this.boundTextField(keySet('company_name'), 'Company Name')}
+        </Cell>
+        <Cell col={4}>
+          {this.boundCountrySelectField(keySet('state_or_territory'), keySet('country'), 'Country')}
+        </Cell>
+        <Cell col={4}>
+          {this.boundStateSelectField(keySet('state_or_territory'), keySet('country'), 'State or Territory')}
+        </Cell>
+        <Cell col={4}>
+          {this.boundTextField(keySet('city'), 'City')}
+        </Cell>
+        <Cell col={12}>
+          {this.boundSelectField(keySet('industry'), 'Industry', this.industryOptions)}
+        </Cell>
+        <Cell col={12}>
+          {this.boundTextField(keySet('position'), 'Position')}
+        </Cell>
+        <Cell col={6}>
+          {this.boundDateField(keySet('start_date'), 'Start Date', true)}
+        </Cell>
+        <Cell col={6}>
+          {this.boundDateField(keySet('end_date'), 'End Date', true)}
+          <span className="end-date-hint">
+            Leave blank if this is a current position
+          </span>
+        </Cell>
+      </Grid>
+    );
+  }
+
+  renderWorkHistory () {
+    const { ui, profile: { work_history } } = this.props;
+    if ( ui.workHistoryEdit === true ) {
+      let workHistoryRows = [];
+      if ( !_.isUndefined(work_history) ) {
+        workHistoryRows = work_history.filter(entry =>
+          entry.id !== undefined
```

Comment:
The `filter` will mess up the indexes which will cause problems with deleting and editing here (although in this case the user will probably be unable to click these buttons while the dialog is open). Can you use `Object.entries` here similar to in EducationTab to preserve the correct index value?